lib/jekyll-admin/server.rb

```diff
@@ -75,7 +75,11 @@ def write_file(path, content)
       File.open(path, "wb") do |file|
         file.write(content)
       end
-      site.process
+      if ENV["RACK_ENV"]
+        site.process
+      else
+        site.read
+      end
```

Comment:
I'm trying try to eliminate the overlapping parts between two threads (regeneration and api server).

Does incremental regeneration call any methods from `site.process` ? If so, calling them again after writing to a file might cause a confusion. Do we really need to call whole methods of site.process, rather than just a few ones that are really necessary for the API ? Because when I use only `site.read`, everything seems to work.

The `if` statement there is for tests to pass.